src/oc/web/actions/app_update.cljs

```diff
@@ -0,0 +1,62 @@
+(ns oc.web.actions.app-update
+  (:require [oc.web.api :as api]
+            [oc.web.actions.notifications :as notification-actions]))
+
+(defonce app-update-cicle (atom false))
+(defonce app-update-timeout (atom nil))
+(defonce should-check-for-updates (atom true))
+
+(def initial-app-update-wait (* 1000 10)) ;; 10 seconds
+(def default-app-update-interval (* 1000 60 5)) ;; 5 minutes
+(def default-app-update-dismiss-interval (* 1000 60 60 24)) ;; 24 hours
+
+(declare real-app-update-check)
+
+(defn- set-app-update-timeout
+  "Set a timeout to check for updates."
+  [interval]
+  (when @app-update-timeout
+    (.clearTimeout js/window @app-update-timeout))
+  (reset! app-update-timeout
+   (.setTimeout js/window real-app-update-check interval)))
+
+(defn- app-update-notification-dismissed
+  "App update notification was dismissed, let's set a new timeout with a longer wait time."
+  []
+  ;; Clear the old interval
+  (when @app-update-timeout
+    (.clearTimeout js/window @app-update-timeout))
+  ;; Set a new interval
+  (set-app-update-timeout default-app-update-dismiss-interval))
+
+(def update-verbage
+  (if js/window.isDesktop
```

Comment:
@bago2k4 as of #972 this will no longer be referred to as `js/window.isDesktop`, but we should instead be checking for the existence of the `js/window.OCCarrotDesktop` object. #972 contains changes that scopes Electron's JS injections under this object in order to avoid any future collisions on the `window` object.

Reply:
We can likely add this PR as a dependency for #972, which would allow us to make the fix in #972 instead of in this branch.